application/ledger_query_service/src/bitcoin/queries.rs

```diff
@@ -1,47 +1,44 @@
-use super::block_processor::{Block, Transaction};
 use crate::{
     query_result_repository::QueryResult,
     route_factory::{Error, ExpandResult, QueryParams, QueryType, ShouldExpand},
-    IsEmpty, Query, QueryMatchResult,
+    IsEmpty,
 };
 use bitcoin_rpc_client::{BitcoinCoreClient, BitcoinRpcApi};
 use bitcoin_support::{
-    serialize::BitcoinHash, Address, MinedBlock as BitcoinBlock, OutPoint, SpendsFrom,
-    SpendsFromWith, SpendsTo, SpendsWith, Transaction as BitcoinTransaction, TransactionId,
+    Address, MinedBlock as Block, OutPoint, SpendsFrom, SpendsFromWith, SpendsTo, SpendsWith,
+    Transaction, TransactionId,
 };
 use futures::Future;
 use std::sync::Arc;
 
 #[derive(Serialize, Deserialize, Clone, Default, Debug)]
-pub struct BitcoinTransactionQuery {
+pub struct TransactionQuery {
     pub to_address: Option<Address>,
     pub from_outpoint: Option<OutPoint>,
     pub unlock_script: Option<Vec<Vec<u8>>>,
-    #[serde(default = "default_confirmations")]
-    confirmations_needed: u32,
 }
 
-impl QueryType for BitcoinTransactionQuery {
+impl QueryType for TransactionQuery {
     fn route() -> &'static str {
         "transactions"
     }
 }
 
-impl ShouldExpand for BitcoinTransactionQuery {
+impl ShouldExpand for TransactionQuery {
     fn should_expand(query_params: &QueryParams) -> bool {
         query_params.expand_results
     }
 }
 
-impl ExpandResult for BitcoinTransactionQuery {
+impl ExpandResult for TransactionQuery {
     type Client = BitcoinCoreClient;
-    type Item = BitcoinTransaction;
+    type Item = Transaction;
 
     fn expand_result(
         result: &QueryResult,
         client: Arc<BitcoinCoreClient>,
-    ) -> Result<Vec<BitcoinTransaction>, Error> {
-        let mut expanded_result: Vec<BitcoinTransaction> = Vec::new();
+    ) -> Result<Vec<Transaction>, Error> {
+        let mut expanded_result: Vec<Transaction> = Vec::new();
```

Comment:
🤡 This type annotation wouldn't be needed if we wrote this as a map.

Reply:
It's actually not needed at all - compiles without it.